TACC/Core-Styles
# feat: s-scrollbars

## Overview

Let users add class `s-scrollbars` to style child element scrollbars.

## Related

- [TUP-414](https://jira.tacc.utexas.edu/browse/TUP-414)

## Changes

- **added** `s-scrollbars`

## Testing

1. Open http://localhost:3000/components/detail/s-scrollbars.
2. Set operating system to auto-hide scrollbars.
	- On **macOS**, set "Show scroll bars" to "**When scrolling**".
	- On **Windows**, enable "Automatically hide scroll bars on Windows".
3. Test scrolling and clicking styled scrollbar on:
	- On browsers that support `::-webkit-scrollbar`.
		- scrollbar presence might ignore user Operating System.
	- On browsers that support `scrollbar-color`
		- scrollbar presence should respect user Operating System.

## UI

### MacOS (only show scrollbars when scrolling) - Old CSS Solution

https://github.com/TACC/Core-Styles/assets/62723358/1514997c-1088-46d9-a035-6dfee0229fac

> **Important**
> OS should only show while scrolling, but styled scrollbars are visible.

> **Warning**
> Designer wants to override OS like this, but this is non-standard CSS.

<details><summary>browser support (not Firefox)</summary><img width="670" alt="old css support" src="https://github.com/TACC/Core-Styles/assets/62723358/0c29befd-cfa0-415b-a0c7-f8b96a19a52d"></details>

### MacOS (only show scrollbars when scrolling) - New CSS Solution

https://github.com/TACC/Core-Styles/assets/62723358/dcb11bef-6db0-47b3-ba89-e57105bb3069

> **Important**
> OS should only show while scrolling, and styled scrollbars obey that.

<details><summary>browser support (only Firefox)</summary><img width="670" alt="new css support" src="https://github.com/TACC/Core-Styles/assets/62723358/79e2124b-f7f5-45ef-a6ed-9dda1c623290"></details>
